src/wast-parser.cc

```diff
@@ -608,7 +608,7 @@ TokenType WastParser::Peek(size_t n) {
       }
       if ((options_->features.code_metadata_enabled() &&
```

Comment:
Seems like maybe there's enough conditions here that they would benefit by getting broken out into a separate function at this point?

Reply:
Agreed. Honestly, I was a bit surprised to see this logic in the `Peek` method, too.  Perhaps it would be better to put this all in it's own method anyway, with a more straightforward way of parsing/skipping certain annotations.